### ISSUE TITLE
Map view accessibility improvements

### ISSUE THREAD

**Issue by bronzehedwick** [MukurtuCMS/Mukurtu-CMS#722]
The Leaflet library has some accessibility issues effecting our implementation, outlined below. Almost all have issues opened upstream. Original comment generating this issue: https://github.com/MukurtuCMS/Mukurtu-CMS/issues/343#issuecomment-2725446495

## Problems

1. If there is more than one marker, and a marker is opened, pressing <kbd>Tab</kbd> moves to the next marker, not the contents of the preview. I would expect the preview to be the next item in the order. Instead, it appears at the end of marker list. https://github.com/Leaflet/Leaflet/issues/8115
2. When the preview is open, pressing <kbd>Escape</kbd> should close the preview. Currently it does nothing. https://github.com/Leaflet/Leaflet/issues/6551
3. Markers should use the `aria-expanded` and `aria-controls` properties for markers and previews. https://github.com/Leaflet/Leaflet/issues/7527
4. The previews lack semantics, which can negatively impact screen readers. Fixing this could fix the <kbd>Escape</kbd> issue (above) at the same time. https://github.com/Leaflet/Leaflet/issues/7968 
5. When navigating to the map, VO announces it as a generic group; it should have a more descriptive label. Suggestion: “[Page title] map”.
6. When navigating to map markers, VO announces them all as “button”. They should have a descriptive label of their contents. This could be at least partially solved by adding semantics (see above).

There’s a whole category of accessibility issues reported against Leaflet, [tracked in Github](https://github.com/Leaflet/Leaflet/issues?q=is%3Aissue%20state%3Aopen%20accessibility%20label%3Aaccessibility). Many of these issues have been open for a while. The maintainer lives in Ukraine, hence understandably has other priorities.

**Comment by bronzehedwick** [comment]
## Proposed solution method

@michael-wynne-wsu Continuing the conversation from https://github.com/MukurtuCMS/Mukurtu-CMS/issues/343#issuecomment-2725446495

Many of these issues are likely only possible to resolve upstream; a proven model is to fork the repository, make the improvements, use that version and contribute our fixes back. Then, if/when our contributions are accepted, or the issues are otherwise fixed, we can switch back to the regular Leaflet release. The downside is that there we will have to maintain this fork for the duration of its lifetime, monitoring the library pulling in any upstream changes as they come in. However, I think it’s the best method of getting the accessibility improvements made in a permanent, sustainable, and maintainable way.

Thoughts?

**Comment by michael-wynne-wsu** [comment]
@bronzehedwick thank you SO much for the thorough review and explanation.

For the scope of our work right now, I don't think we want to sink your hours on this and I'd like to maintain this as an issue that we revisit in the future. Let me know if you have concerns about that though.

**Comment by bronzehedwick** [comment]
Yep, that totally makes sense @michael-wynne-wsu. I think these are clear issues, but they are largely tracked upstream, so have a boarder surface of developers to address them. I'll unassigned this from myself.

**Comment by michael-wynne-wsu** [comment]
Agreed @bronzehedwick that's all spot on, thank you!
